### PR TITLE
CLID-53: Introduce env var to provide OCP release signature verificat…

### PR DESCRIPTION
**Pull request by sherine-k**
# Description

Introduce env var to provide OCP release signature verification key

Fixes # [CLID-53](https://issues.redhat.com//browse/CLID-53)

:warning: **The commented unit test has shown that oc-mirror doesn't stop when signature is not verified**: Please advise if this is the requested behavior.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- New unit tests have been introduced. 
- Execute oc-mirror mirror to disk with the following env var exported - logs should not show errors:
```
OCP_SIGNATURE_VERIFICATION_PK=./tests/custom-ocp-sig-key.asc
```
- Execute oc-mirror mirror to disk with the following env var exported - logs should show errors but not fail:
```
OCP_SIGNATURE_VERIFICATION_PK=./tests/buggy-ocp-sig-key.asc
```
```
2024/04/18 17:41:33  [ERROR]  : unexpected EOF
2024/04/18 17:42:25  [ERROR]  : invalid signature
2024/04/18 17:42:41  [ERROR]  : unexpected openpgp.MessageDetails: neither Signature nor SignatureV3 is set
```

### CODE REVIEW

File: v2/pkg/release/signature.go [hunk 1]
```diff
@@ -25,10 +25,26 @@ type SignatureSchema struct {
...
 	Config v1alpha2.ImageSetConfiguration
 	Opts   mirror.CopyOptions
+	pgpKey string
 }
 
 func NewSignatureClient(log clog.PluggableLoggerInterface, config v1alpha2.ImageSetConfiguration, opts mirror.CopyOptions) SignatureInterface {
-	return &SignatureSchema{Log: log, Config: config, Opts: opts}
+	var pgp string
+	if pgpKeyOverride := os.Getenv("OCP_SIGNATURE_VERIFICATION_PK"); len(pgpKeyOverride) != 0 {
+		log.Info("OCP_SIGNATURE_VERIFICATION_PK environment variable set: using PGP key in %s for OCP signature verification", pgpKeyOverride)
```
Comment (aguidirh) [inline v2/pkg/release/signature.go:34]: From my perspective oc-mirror should show only high level information as Info, signature details is something someone is going to verify only when there is something wrong (Debug Level).

Comment (aguidirh) [inline v2/pkg/release/signature.go:34]: It is good to have it as an env var, thanks.

**Review by aguidirh — COMMENTED** [review]
Thanks for this change @sherine-k. 

Only a small suggestion (nit).

**Comment by lmzuccarelli** [comment]
@sherine-k - make the change and we can get this merged asap. Happy to give it  looks good to me :)

**Pull request merged**